codeflash/code_utils/env_utils.py

```diff
@@ -1,11 +1,24 @@
 import os
+import subprocess
 from functools import lru_cache
 from typing import Optional
 
 from codeflash.cli_cmds.console import logger
 from codeflash.code_utils.shell_utils import read_api_key_from_shell_config
 
 
+def check_formatter_installed(formatter_cmds: list[str]) -> bool:
+    cmd_parts = formatter_cmds[0].split(" ")
+    if "black" in cmd_parts or "ruff" in cmd_parts:
+        formatter = cmd_parts[0]
```

Comment:
this logic will fail when someone has setup `uv run black $file` 
the `in` condition will match and the `cmd_parts[0]` will be `uv` which will pass even if black is not present.
Make it run the sub command until black is detected, so if black is index 2 in the split list then run `cmd_parts[:2+1]`